lib/train/k8s/container/kubectl_exec_client.rb

```diff
@@ -8,12 +8,10 @@ module Container
       class KubectlExecClient
         attr_reader :pod, :container_name, :namespace
 
-        DEFAULT_NAMESPACE = "default".freeze
-
         def initialize(pod:, namespace: nil, container_name: nil)
           @pod = pod
           @container_name = container_name
-          @namespace = namespace || DEFAULT_NAMESPACE
+          @namespace = namespace
```

Comment:
so the default namespace logic is kept here because it is the responsibility of the kubectl client to be aware of the connection internal details of the cluster.. so I would keep this logic in here

Reply:
But isn't that good to keep all connection parameters together with default values? When I look at the code I am expecting if namespace OR host is not provided it should assign default namespace before connection happens.  I had to run through all flow to understand if those two parameters are not available then what's the default getting assigned.

Reply:
yeah because it is the responsibility of the K8s client .. not train transport connection

Reply:
one thing if you want to do like this https://github.com/inspec/train-k8s-container/pull/17/files#diff-c4540897f7e17865b4c727a6a2549a98746fa379c627f8cd12cc248b14e1d1cbR20, 

then you can do `host || Train::K8s::Container::KubectlExecClient::DEFAULT_NAMESPACE`